code/modules/asset_cache/asset_list.dm

```diff
@@ -5,18 +5,31 @@
 GLOBAL_LIST_EMPTY(asset_datums)
 
 //get an assetdatum or make a new one
-/proc/get_asset_datum(type)
+//does NOT ensure it's filled, if you want that use get_asset_datum()
+/proc/load_asset_datum(type)
 	return GLOB.asset_datums[type] || new type()
 
+/proc/get_asset_datum(type)
+	var/datum/asset/loaded_asset = GLOB.asset_datums[type] || new type()
```

Comment:
Is it worth saving this new value to cache if it doesn't exist?

Reply:
Not sure, just copied that from TG.
It just works™️

Works pretty good btw